fusion.md

```diff
@@ -0,0 +1,1396 @@
+<!-- XXX: I'm not sure what we are trying to say here -->
+
+Linking in your feedbacks
+-------------------------
+TODO:
+
+Intended Use of GCAM Fusion
+---------------------------
+
+Note that GCAM Fusion gives the users full access to all the internal parameters
+of GCAM for better or for worse.  Just because you are able to change these
+values doesn't mean GCAM will be able to operate normally when doing so. 
+Therefore we only reccommend using GCAM Fusion inside of the `IModelFeedback`
+methods.  Making feedbacks during the solution of a model period would require
+additional dependencies and linkages to ensure proper solution and GCAM Fusion
+would entirely circumvent those procedures. 
+As a rule of thumb adjusting the same model perameters which are parsed in GCAM
+XML input files should be fine to modify.  It should **not** be used to
+curcimvent normal object orientened principals or designs.  Object encapsulation
+allows us to ensure some level of consistency. 
+
+To be clear there are no sofware limitation imposed on the use of GCAM Fusion
+however code proposed for inclusion into the Core GCAM model may be rejected due
+to improper / abuse of the capabilities as it will hinder the long term
+maintainability of the model.
+
+
+Some Documentation for Filter Steps
+---
+
+### FilterStep objects
+
+
+
+A FilterStep is the object that GCAMFusion uses to search a single GCAM
+container's data vector.  It can optionally specify a data name to match which
+is compared against the `Data::mDataName`.  If the data name in the FilterStep
+is empty it is assumed to match *any* data name.  
+
+The other optional parameter is a [Filter](#filter-objects).  Filter objects are
+valid for search targets that are containers for other objects.  Such containers
+are indicated with the `ARRAY` or `CONTAINER` flag.  If specified, a filter can
+be used to select any single element of the matched container.  If no filter is
+specified, it is assumed to be `NoFilter`, which selects the entire
+container.  Note that if a Filter other than `NoFilter` is set, and the matched
+object is not a container (i.e., has the `SIMPLE` flag, then the match will be
+rejected even if the data name matches.  
+
+In addition if no data name and no filter is set then not only does this
+FilterStep match all data but it also enables special "descendant step"
+traversal behavior in GCAMFusion where the next FilterStep can be matched zero
+or more containers down.  This is analogous to the `//` operator in XPath
+queries.  For example, if a 'sector' object has 'subsector' children, which in
+turn have 'technology' children of their own, then `sector//share-weight` will
+find data named 'share-weight' at both the subector and technology levels.  It
+would also find a share-weight object contained in the sector itself if there
```

Comment:
I tried to clarify this a little.  Is what I wrote on this line accurate?

Reply:
Yup, this is accurate